src/animation_player.rs

```diff
@@ -181,26 +241,31 @@ pub fn draw_movement_arrows(mut gizmos: Gizmos, mut event_reader: EventReader<Hi
     }
 }
 
-#[derive(Resource)]
-pub struct Time {
-    pub time: f32,
-}
-
-pub fn frame_interpolation(
-    mut commands: Commands,
-    // q_names: Query<(Entity, &Name, &Transform)>,
-    mut bvh_data: ResMut<BvhData>,
-    // time: Res<Time>,
-) {
-    if let Some(bvh_vec) = &bvh_data.bvh_animation {
-        let bvh: &Bvh = &bvh_vec[1];
+pub fn get_pose(
+    mut time: ResMut<Time>,
+    bvh_data: &mut ResMut<BvhData>,
+    mut event_reader: EventReader<FrameUpdateEvent>,
+) -> f32 {
+    let duration_per_frame = 0.033333;
+    // let total_duration = num_frames as f32 * duration_per_frame;
+    let mut frame_index: usize = (time.time / duration_per_frame).floor() as usize;
 
-        let duration_per_frame = 0.033333;
-        let total_duration = duration_per_frame * bvh.frames().len() as f32;
-        // let frame_index = (time.time / duration_per_frame).floor();
+    println!("Current Frame Index: {}", frame_index);
 
-        if let Some(frame) = bvh.frames().nth(0) {
-            println!("Frame 0 {:?}", frame);
+    for event in event_reader.read() {
+        if time.time != event.time {
+            time.time = event.time;
+            frame_index = (time.time / duration_per_frame).floor() as usize;
+            bvh_data.current_frame_index = frame_index;
         }
     }
+    // Find how much time has passed since the last frame (finds the remainder)
+    // Calculate how far this is through the frame (divides the remainder)
+    let factor: f32 = (time.time % duration_per_frame) / duration_per_frame;
+
+    return factor;
+}
+
+pub fn animation_time(mut event_writer: EventWriter<FrameUpdateEvent>) {
+    event_writer.send(FrameUpdateEvent { time: 30.0 });
```

Comment:
I would recommend using Bevy's [`bevy::time::Time::elapsed_seconds`](https://docs.rs/bevy/latest/bevy/time/struct.Time.html#method.elapsed_seconds) instead.

Note, it is already included in `bevy::prelude::*` so no need to import anything extra.